### PR TITLE
Break out LM317_3PinPackage into unique symbols

### PR DESCRIPTION
**Pull request by evanshultz**
Oddly, there was one LM317 symbol for all 3-pin packages. And it had a terribly ugly name. LM337 had separate symbols for each package. So this splits up the LM317 symbol similarly.

In some cases the TI doc referenced didn't cover the specific package, so I had to find new datasheets. I stuck with TI. And the Diodes Inc datasheet changed and even in NRND status I kept it. Here are all the datasheets I used:
- http://www.ti.com/lit/ds/symlink/lm317.pdf (SOT-223, TO-220, TO-263)
- http://www.ti.com/lit/ds/symlink/lm317l.pdf (SOT-89 [changed from LM317 to LM317L])
- http://www.ti.com/lit/ds/snvs774n/snvs774n.pdf (TO-252)
- https://www.diodes.com/assets/Datasheets/products_inactive_data/AZ1117.pdf

The TO-252 clearly had the middle pin missing, so I went with that footprint. The TO-263 from both TI and Diodes Inc shows 4 pins in the package drawing so I used the matching footprint there.

Here is one reference symbol just for your enjoyment, though they're all direct copies of the seed symbol with only metadata changed:
image

I will submit another PR to remove the chimera symbol for v6's release.

---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [ ] Give a reason behind any intentional library convention rule violation.

### CODE REVIEW

**Comment by chschlue** [comment]
I'm not sure about the TO-263 one.
ON is pretty clear on that their D²PAK LM317 comes with a vestigial pin 2 (https://www.onsemi.com/pub/Collateral/LM317-D.PDF). Same for ST (https://www.st.com/resource/en/datasheet/lm317.pdf).
TI claims not to offer a 2-pin TO-252 at all but has a 2-pin TO-263 (https://www.ti.com/packaging/docs/searchtipackages.tsp?packageName=TO) with no parts using it (https://www.ti.com/packaging/docs/searchproductbypackage.tsp?orderablePartNumber=&packageDesignator=KTT&pinCount=2&results=results#resultspage).

I tend to think TI messed up and we should go with Package_TO_SOT_SMD:TO-263-2 by default.

**Comment by evanshultz** [comment]
@chschlue 
Alrighty. Done!

File: Regulator_Linear.dcm [hunk 1]
```diff
@@ -2754,6 +2784,12 @@ K Adjustable Voltage Regulator 100mA Positive
 F http://www.ti.com/lit/ds/snvs775k/snvs775k.pdf
 $ENDCMP
 #
+$CMP LM317L_SOT-89
+D 1A 35V Adjustable Linear Regulator, SOT-89
```
Comment (chschlue) [inline Regulator_Linear.dcm:2788]: Should be `100mA`

File: Regulator_Linear.dcm [hunk 2]
```diff
@@ -2766,6 +2802,30 @@ K Adjustable Voltage Regulator 1A Positive
 F http://www.ti.com/lit/ds/symlink/lm317.pdf
 $ENDCMP
 #
+$CMP LM317_SOT-223
+D 1A 35V Adjustable Linear Regulator, SOT-223
```
Comment (chschlue) [inline Regulator_Linear.dcm:2806]: `1.5A`

File: Regulator_Linear.dcm [hunk 3]
```diff
@@ -2766,6 +2802,30 @@ K Adjustable Voltage Regulator 1A Positive
...
+K Adjustable Voltage Regulator 1A Positive
+F http://www.ti.com/lit/ds/symlink/lm317.pdf
+$ENDCMP
+#
+$CMP LM317_TO-220
+D 1A 35V Adjustable Linear Regulator, TO-220
```
Comment (chschlue) [inline Regulator_Linear.dcm:2812]: `1.5A`

File: Regulator_Linear.dcm [hunk 4]
```diff
@@ -2766,6 +2802,30 @@ K Adjustable Voltage Regulator 1A Positive
...
+K Adjustable Voltage Regulator 1A Positive
+F http://www.ti.com/lit/ds/symlink/lm317.pdf
+$ENDCMP
+#
+$CMP LM317_TO-252
+D 1A 35V Adjustable Linear Regulator, TO-252
```
Comment (chschlue) [inline Regulator_Linear.dcm:2818]: `1.5A`

File: Regulator_Linear.dcm [hunk 5]
```diff
@@ -2766,6 +2802,30 @@ K Adjustable Voltage Regulator 1A Positive
...
+K Adjustable Voltage Regulator 1A Positive
+F http://www.ti.com/lit/ds/snvs774n/snvs774n.pdf
+$ENDCMP
+#
+$CMP LM317_TO-263
+D 1A 35V Adjustable Linear Regulator, TO-263
```
Comment (chschlue) [inline Regulator_Linear.dcm:2824]: `1.5A`

**Comment by evanshultz** [comment]
I had assumed that the previous description was correct and just needed to split the packages. Probably should have caught LM317L but otherwise how stupid to assume. Thanks for catching that!

**Pull request merged**